core-java-modules/core-java-string-conversions-2/src/test/java/com/baeldung/stringtocamelcase/StringtocamelcaseApplicationTests.java

```diff
@@ -0,0 +1,61 @@
+package com.baeldung.stringtocamelcase;
+
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+
+public class StringtocamelcaseApplicationTests {
+
+	private static final String TEXT_WHITESPACE = "THIS STRING SHOULD BE IN CAMEL CASE";
+	private static final String TEXT_UNDERSCORE = "THIS_STRING_SHOULD_BE_IN_CAMEL_CASE";
+	private static final String TEXT_EXPECTED = "ThisStringShouldBeInCamelCase";
+
+	@Test
+	public void testNothing() {
+	}
+
+	@Test
+	public void testToCamelCaseByIteration_shouldReturnCamelCase() {
+		assertEquals(TEXT_EXPECTED, StringtocamelcaseApplication.toCamelCaseByIteration(TEXT_WHITESPACE, ' '));
+		assertEquals(TEXT_EXPECTED, StringtocamelcaseApplication.toCamelCaseByIteration(TEXT_UNDERSCORE, '_'));
+	}
+
+	@Test
+	public void testToCamelCaseBySplitting_shouldReturnCamelCase() {
+		assertEquals(TEXT_EXPECTED, StringtocamelcaseApplication.toCamelCaseBySplitting(TEXT_WHITESPACE, " "));
+		assertEquals(TEXT_EXPECTED, StringtocamelcaseApplication.toCamelCaseBySplitting(TEXT_UNDERSCORE, "_"));
+	}
+
+	@Test
+	public void testToCamelCaseBySplittingUsingStreams_shouldReturnCamelCase() {
```

Comment:
These splitting cases... if you get the regular expression right, can't they also convert titlecase to camel case? It's a question of which regular expression you put into the `delimiter`. String.split uses a regex.